libbpf-rs/src/program.rs

```diff
@@ -42,11 +51,20 @@ impl OpenProgram {
         }
     }
 
-    /// Name of the section this `Program` belongs to.
+    /// Name of the section this `OpenProgram` belongs to.
     pub fn section(&self) -> &str {
         &self.section
     }
 
+    /// The name of this `OpenProgram`.
+    pub fn name(&self) -> Result<&str> {
+        let name_ptr = unsafe { libbpf_sys::bpf_program__name(self.ptr) };
+        let name_c_str = unsafe { CStr::from_ptr(name_ptr) };
+        name_c_str
+            .to_str()
+            .map_err(|e| Error::Internal(e.to_string()))
```

Comment:
Nit: Should we use `c_ptr_to_string` here?

Reply:
we could; the main difference is that `c_ptr_to_string` returns a `Result<String>` and this returns `Result<&str>` which doesn't copy.
LMK which do you prefer.

Reply:
Ah, true. No need to allocate. Let's keep what you have.